core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -302,8 +281,21 @@ namespace kagome::consensus {
   }
 
   void BabeImpl::finishSlot() {
+    if (not slots_leadership_.has_value()) {
+      auto &&[best_block_number, best_block_hash] = block_tree_->deepestLeaf();
+
+      auto epoch_res = block_tree_->getEpochDescriptor(
+          current_epoch_.epoch_index, best_block_hash);
+      BOOST_ASSERT(epoch_res.has_value());
+      auto &epoch = epoch_res.value();
```

Comment:
And here, is it meant to throw an exception in release?

Reply:
It's ok. For any epoch will be returned at least initial epoch digest. Possible just one way to get failure - is absenting block with best_block_hash